src/pages/iou/request/step/IOURequestStepMerchant.js

```diff
@@ -70,7 +101,24 @@ function IOURequestStepMerchant({
      * @param {String} value.moneyRequestMerchant
      */
     const updateMerchant = (value) => {
-        IOU.setMoneyRequestMerchant_temporaryForRefactor(transactionID, value.moneyRequestMerchant);
+        const newMerchant = value.moneyRequestMerchant.trim();
+        // In the split flow, when editing we use SPLIT_TRANSACTION_DRAFT to save draft value
+        if (isEditingSplitBill) {
+            IOU.setDraftSplitTransaction(transactionID, {merchant: newMerchant});
+            navigateBack();
+            return;
+        }
+
+        // In case the merchant hasn't been changed, do not make the API request.
+        // In case the merchant has been set to empty string while current merchant is partial, do nothing too.
+        if (newMerchant === merchant || (newMerchant === '' && merchant === CONST.TRANSACTION.PARTIAL_TRANSACTION_MERCHANT)) {
+            navigateBack();
+            return;
+        }
+        IOU.setMoneyRequestMerchant(transactionID, newMerchant, !isEditing);
+        if (isEditing) {
+            IOU.updateMoneyRequestMerchant(transactionID, reportID, newMerchant || CONST.TRANSACTION.PARTIAL_TRANSACTION_MERCHANT, policy, policyTags, policyCategories);
```

Comment:
Why do we need the `|| CONST.TRANSACTION.PARTIAL_TRANSACTION_MERCHANT` fallback? Can you add a comment explaining it?

Reply:
Migrate from here

https://github.com/Expensify/App/blob/1f1867a0a969c91e2169603b33ea272067428504/src/pages/EditRequestPage.js#L143-L150

Reply:
Gotcha. Can you add a comment explaining why it's needed?

Reply:
> Why do we need the || CONST.TRANSACTION.PARTIAL_TRANSACTION_MERCHANT fallback?

I think it is our design

Reply:
How about this, which is based on your other comment:
```
// When creating new money requests newMerchant can be blank so we fall back on PARTIAL_TRANSACTION_MERCHANT
```

Ideally it should also say *why* we need the fallback, but I don't know the reason for not setting merchant to be blank here. Do you?

Reply:
Similar to you 😄